### PR TITLE
fixes for activity log (#1574)

### PR DESCRIPTION
**Pull request by saravanpa-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/6702

*Description of changes:*

Changes for recording  user id in activity log table
added masking for bc reg non-staff users

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-api/migrations/versions/3c89ccb6bb32_actor_changed_to_actor_id.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# revision identifiers, used by Alembic.
+revision = '3c89ccb6bb32'
+down_revision = 'd0392ebda924'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.drop_column('activity_logs', 'actor')
+    op.add_column('activity_logs', sa.Column('actor_id', sa.Integer(), nullable=True))
```
Comment (karthik-aot) [inline auth-api/migrations/versions/3c89ccb6bb32_actor_changed_to_actor_id.py:21]: foreign_key to user?

**Review by karthik-aot — APPROVED** [review]

File: auth-api/src/auth_api/resources/acitivity_log.py [hunk 2]
```diff
@@ -36,7 +36,8 @@ class ActivityLog(Resource):
     @staticmethod
     @TRACER.trace()
     @cors.crossdomain(origin='*')
-    @_jwt.has_one_of_roles([Role.STAFF.value])
+    @_jwt.has_one_of_roles([Role.SYSTEM.value, Role.STAFF.value,
+                            Role.PUBLIC_USER.value, Role.GOV_ACCOUNT_USER.value])
```
Comment (sumesh-aot) [inline auth-api/src/auth_api/resources/acitivity_log.py:40]: you can replace PUBLIC_USER and GOV_ACCOUNT_USER by ACCOUNT_HOLDER

**Review by sumesh-aot — APPROVED** [review]
Please check comment about role

File: auth-api/migrations/versions/3c89ccb6bb32_actor_changed_to_actor_id.py (hunk 1 above)
Comment (saravanpa-aot) [reply auth-api/migrations/versions/3c89ccb6bb32_actor_changed_to_actor_id.py:21]: I am not changing this now..but will keep this in mind and see in next sprint as well if we can do that.

**Pull request merged**